parcels/particle.py

```diff
@@ -36,6 +37,57 @@ def AdvectionEE(particle, grid, time, dt):
     particle.lat += v1 * dt
 
 
+def AdvectionRK45(particle, grid, time, dt):
+    tol = [1e-9]
```

Comment:
Technically yes, but it breaks all the current abstractions we have. By that I mean it shouldn't really be in the `execute` interface in the first place, since it has no meaning for any other algorithm. It also doesn't seem to have any effect on the current tests so I would leave this until someone re-enables the Stommel test and implements appropriate acceptance criteria that highlight the sensitivity to `tol`.

In the long term, one really cool, but complicated way to do this would be to have smart kernel-factories that generate parameterisable versions of known kernels on-the-fly, for example `AdvectionRKN(N=8, tol=1.e-6)`. A somewhat easier approach would be to have a global "parameter dictionary" to expose these algorithm-specific options and thus allow users to override them. However, none of these suggestions are feasible with the current infrastructure, so I just hard-coded it to get this branch merged.